invesalius/data/viewer_volume.py

```diff
@@ -520,41 +525,89 @@ def OnSensors(self, markers_flag):
         else:
             colour3 = (1, 0, 0)
 
-        self.probe.SetColour(colour1)
-        self.ref.SetColour(colour2)
-        self.obj.SetColour(colour3)
+        self.dummy_probe_actor.GetProperty().SetColor(colour1)
+        self.dummy_ref_actor.GetProperty().SetColor(colour2)
+        self.dummy_obj_actor.GetProperty().SetColor(colour3)
 
     def CreateSensorID(self):
-        probe = vtku.Text()
-        probe.SetSize(const.TEXT_SIZE_LARGE)
-        probe.SetPosition((const.X, const.Y))
-        probe.ShadowOff()
-        probe.SetValue("P")
-        self.probe = probe
-        self.ren.AddActor(probe.actor)
-
-        ref = vtku.Text()
-        ref.SetSize(const.TEXT_SIZE_LARGE)
-        ref.SetPosition((const.X+0.04, const.Y))
-        ref.ShadowOff()
-        ref.SetValue("R")
-        self.ref = ref
-        self.ren.AddActor(ref.actor)
-
-        obj = vtku.Text()
-        obj.SetSize(const.TEXT_SIZE_LARGE)
-        obj.SetPosition((const.X+0.08, const.Y))
-        obj.ShadowOff()
-        obj.SetValue("O")
-        self.obj = obj
-        self.ren.AddActor(obj.actor)
-        self.UpdateRender()
+        self.ren_probe = vtkRenderer()
+        self.ren_probe.SetLayer(1)
+
+        self.interactor.GetRenderWindow().AddRenderer(self.ren_probe)
+        self.ren_probe.SetViewport(0.01, 0.79, 0.15, 0.99)
+        filename = os.path.join(inv_paths.OBJ_DIR, "stylus.stl")
+
+        reader = vtkSTLReader()
+        reader.SetFileName(filename)
+        mapper = vtkPolyDataMapper()
+        mapper.SetInputConnection(reader.GetOutputPort())
+
+        dummy_probe_actor = vtkActor()
+        dummy_probe_actor.SetMapper(mapper)
+        dummy_probe_actor.GetProperty().SetColor(1, 1, 1)
+        dummy_probe_actor.GetProperty().SetOpacity(1.)
+        self.dummy_probe_actor = dummy_probe_actor
+
+        self.ren_probe.AddActor(dummy_probe_actor)
+        self.ren_probe.InteractiveOff()
+
+        self.ren_ref = vtkRenderer()
+        self.ren_ref.SetLayer(1)
+
+        self.interactor.GetRenderWindow().AddRenderer(self.ren_ref)
+        self.ren_ref.SetViewport(0.01, 0.59, 0.15, 0.79)
+        filename = os.path.join(inv_paths.OBJ_DIR, "head.stl")
+
+        reader = vtkSTLReader()
+        reader.SetFileName(filename)
+        mapper = vtkPolyDataMapper()
+        mapper.SetInputConnection(reader.GetOutputPort())
+
+        dummy_ref_actor = vtkActor()
+        dummy_ref_actor.SetMapper(mapper)
+        dummy_ref_actor.GetProperty().SetColor(1, 1, 1)
+        dummy_ref_actor.GetProperty().SetOpacity(1.)
+        self.dummy_ref_actor = dummy_ref_actor
+
+        self.ren_ref.AddActor(dummy_ref_actor)
+        self.ren_ref.InteractiveOff()
+
+        self.ren_obj = vtkRenderer()
+        self.ren_obj.SetLayer(1)
+
+        self.interactor.GetRenderWindow().AddRenderer(self.ren_obj)
+        self.ren_obj.SetViewport(0.01, 0.39, 0.15, 0.59)
+        filename = os.path.join(inv_paths.OBJ_DIR, "magstim_fig8_coil.stl")
```

Comment:
I would check if the coil without the handle looks better. That simplifies the icon.

Reply:
Indeed, looks much better without the handle!